moveit_core/trajectory_processing/include/moveit/trajectory_processing/time_optimal_trajectory_generation.h

```diff
@@ -174,13 +174,20 @@ class TimeOptimalTrajectoryGeneration : public TimeParameterization
                          const std::unordered_map<std::string, double>& velocity_limits,
                          const std::unordered_map<std::string, double>& acceleration_limits) const override;
 
+  /**
+   * \brief Compute a trajectory with the given number of waypoints (plus or minus 1, due to rounding)
+   */
+  bool computeTimeStamps(const size_t num_waypoints, robot_trajectory::RobotTrajectory& trajectory,
```

Comment:
So, the way to look at this is just another overloaded version of `computeTimeStamps`. It can be used the same way, just with the additional `num_waypoints` argument.

Reply:
You're right, that is also a way to look at it. I am fine with that :+1: